packages/flutter/test/material/text_field_test.dart

```diff
@@ -505,6 +508,52 @@ void main() {
     );
   }, skip: isBrowser);
 
+  testWidgets('text field toolbar options correctly changes options',
+      (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: Center(
+            child: Container(
+              width: 100,
+              height: 100,
+              child: const Opacity(
+                opacity: 0.5,
+                child: TextField(
+                  decoration: InputDecoration(hintText: 'Placeholder'),
+                  toolbarOptions: ToolbarOptions(copy: true),
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    await tester.showKeyboard(find.byType(TextField));
+
+    const String testValue = 'A B C';
+    tester.testTextInput
+        .updateEditingValue(const TextEditingValue(text: testValue));
+    await tester.pump();
+
+    // The selectWordsInRange with SelectionChangedCause.tap seems to be needed to show the toolbar.
+    // (This is true even if we provide selection parameter to the TextEditingValue above.)
+    final EditableTextState state =
+        tester.state<EditableTextState>(find.byType(EditableText));
+    state.renderEditable.selectWordsInRange(
+        from: const Offset(0, 0), cause: SelectionChangedCause.tap);
+
+    expect(state.showToolbar(), true);
+
+    // This is needed for the AnimatedOpacity to turn from 0 to 1 so the toolbar is visible.
+    await tester.pumpAndSettle();
+    await tester.pump(const Duration(seconds: 1));
+
+    // Sanity check that the toolbar widget exists and does not inclue [Select All].
+    expect(find.text('COPY'), findsOneWidget);
```

Comment:
Can you also check for other context menu? However, be aware we have a bug that select all is always shown which i have a pr up to fix it https://github.com/flutter/flutter/pull/40713#pullrequestreview-289988202
If you encounter this issue, just enable select all in toolbaroptions in this test and check it exists to get around it.

Reply:
done